rotator/rotator.go

```diff
@@ -200,31 +220,34 @@ func (r *Rotator) rotate() error {
 		return err
 	}
 
-	r.wg.Add(1)
-	go func() {
-		err := compress(rotname)
-		if err == nil {
-			os.Remove(rotname)
-		}
-		r.wg.Done()
-	}()
+	if r.newCompressor != nil {
+		r.wg.Add(1)
+		go func() {
+			err := r.compress(rotname)
+			if err == nil {
+				os.Remove(rotname)
+			}
+			r.wg.Done()
+		}()
+	}
 
 	return nil
 }
 
-func compress(name string) (err error) {
+func (r *Rotator) compress(name string) (err error) {
 	f, err := os.Open(name)
 	if err != nil {
 		return err
 	}
 	defer f.Close()
 
-	arc, err := os.OpenFile(name+".gz", os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0644)
+	zname := fmt.Sprintf("%s.%s", name, r.compressSuffix)
+	arc, err := os.OpenFile(zname, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0644)
 	if err != nil {
 		return err
 	}
 
-	z := gzip.NewWriter(arc)
+	z := r.newCompressor(arc)
```

Comment:
With the suggested signature, we'd need an extra `err` check here. To make things safer if the compressor fails to be created, the `return arc.Close()` could be `defer arc.Close()` right before this spot.